### PR TITLE
LPS-198426 Isolate search predicate with parentheses

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-198426

### CODE REVIEW

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java [hunk 1]
```diff
@@ -3091,6 +3133,53 @@ private void _testAddObjectEntryAccountEntryRestriction(
...
 	}
 
+	private void _testGetObjectEntriesWithAccountEntryRestrictedAndSearch()
+		throws Exception {
+
+		_addResourcePermission(ObjectActionKeys.ADD_OBJECT_ENTRY, _buyerRole);
+
+		AccountEntry accountEntry1 = _addAccountEntry();
+
+		PermissionThreadLocal.setPermissionChecker(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:3143]: Maybe you have to store the ThreadLocal original values before set the new ones and then set them back in a finally statement. Can you check how other test classes does that?

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java (hunk 1 above)
Comment (p-albuquerque) [reply modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/manager/v1_0/test/DefaultObjectEntryManagerImplTest.java:3143]: @guilhermedcamacho  Right after this test, there is a moment that the test set another new user in threadLocal, so I believe have no issue about it here

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:forward

**Comment by p-albuquerque** [comment]
ci:forward

**Comment by p-albuquerque** [comment]
directly sent: https://github.com/brianchandotcom/liferay-portal/pull/142100

**Pull request closed without merging**